parser/prism/Translator.cc

```diff
@@ -467,6 +468,31 @@ std::unique_ptr<parser::Node> Translator::translate(pm_node_t *node) {
             auto usedForKeywordArgs = false;
             return translateHash(node, reinterpret_cast<pm_hash_node *>(node)->elements, usedForKeywordArgs);
         }
+        case PM_HASH_PATTERN_NODE: { // An array pattern such as the `[head, *tail]` in the `a in [head, *tail]`
+            auto hashPatternNode = reinterpret_cast<pm_hash_pattern_node *>(node);
+            pm_location_t *loc = &hashPatternNode->base.location;
+
+            auto prismElements = absl::MakeSpan(hashPatternNode->elements.nodes, hashPatternNode->elements.size);
+            auto prismRestNode = hashPatternNode->rest;
+
+            NodeVec sorbetElements{};
+            sorbetElements.reserve(prismElements.size() + (prismRestNode != nullptr ? 1 : 0));
+
+            translateMultiInto(sorbetElements, prismElements);
+            if (prismRestNode != nullptr) {
+                switch (PM_NODE_TYPE(prismRestNode)) {
+                    case PM_ASSOC_SPLAT_NODE: {
+                        sorbetElements.emplace_back(make_unique<parser::MatchRest>(
+                            parser.translateLocation(&prismRestNode->location), nullptr));
+                        break;
+                    }
+                    default:
+                        sorbetElements.emplace_back(translate(prismRestNode));
+                }
```

Comment:
What's the benefit of using a `switch` vs. conditional here? Are we going to add more cases? Or is this a C thing? 😆